File: apps/admin_mobile/lib/application/use_cases/reservations/create_reservation_use_case.dart
import '../../../domain/entities/reservation.dart';
import '../../../domain/repositories/reservation_repository.dart';

class CreateReservationUseCase {
  final ReservationRepository _repo;
  CreateReservationUseCase(this._repo);

  Future<Reservation> call({
    required int clientResourceId,
    required int serviceId,
    required String scheduledAt,
    int? assignedTo,
    String? notes,
  }) =>
      _repo.create(
        clientResourceId: clientResourceId,
        serviceId: serviceId,
        scheduledAt: scheduledAt,
        assignedTo: assignedTo,
        notes: notes,
      );
}

File: apps/admin-v2/src/application/use-cases/reservations/create-reservation.use-case.ts
import type { ReservationRepository, CreateReservationData } from '@/domain/repositories/reservation.repository';

export class CreateReservationUseCase {
  constructor(private repo: ReservationRepository) {}

  execute(data: CreateReservationData) {
    return this.repo.create(data);
  }
}
